### PR TITLE
Feat/#56 sidebar tracking

### PR DESCRIPTION
**Pull request by ciaranschutte**

### CODE REVIEW

**Comment by ciaranschutte** [comment]
closing to address: https://github.com/icgc-argo/platform-ui/pull/1077#issuecomment-584798473

**Comment by ciaranschutte** [comment]
@hlminh2000 changes addressed

File: website/package.json [hunk 1]
```diff
@@ -15,13 +15,14 @@
     "@babel/core": "^7.7.2",
     "@docusaurus/core": "2.0.0-alpha.35",
     "@docusaurus/preset-classic": "2.0.0-alpha.37",
-    "@icgc-argo/uikit": "^1.0.9",
+    "@icgc-argo/uikit": "^1.0.1-0.alpha.ciaran.7",
```
Comment (joneubank) [inline website/package.json]: Should I ask what this version naming scheme is?

File: website/src/components/ContentMenu/index.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    const onscroll = e => {
+      const domElementInFocus = findActiveSection();
+      if (domElementInFocus) {
+        const activeName = domElementInFocus.dataset.menuTitle;
+        setActiveItemName(activeName);
+      }
+    };
+    document.addEventListener('scroll', onscroll);
+    return () => document.removeEventListener('scroll', onscroll);
```
Comment (joneubank) [inline website/src/components/ContentMenu/index.js]: Thank you for remembering to remove old listeners.

File: website/package.json (hunk 1 above)
Comment (ciaranschutte) [reply website/package.json]: @joneubank ah yes! local npm package of uikit isn't working. So to test new uikit changes working in argo docs, just using made up naming for my testing.

Once it's approved in argo-platform repo, uikit should be published using our standard naming

Does this sound ok to you?

Comment (joneubank) [reply website/package.json]: sounds good. i merged the PR in argo-platform just now... hopefully that didn't ruin the flow you had planned 😟

Comment (ciaranschutte) [reply website/package.json]: all good i think!

File: website/src/components/ContentMenu/index.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import Menu from '@icgc-argo/uikit/ContentMenu';
+
+const useMenuHighlight = (schemaRefs, scrollYOffset) => {
+  const [activeItemName, setActiveItemName] = React.useState('');
+
+  React.useEffect(() => {
+    const findActiveSection = () => {
+      const focusedDomEls =
+        schemaRefs.length > 0 &&
+        schemaRefs.filter(ref => {
```
Comment (hlminh2000) [inline website/src/components/ContentMenu/index.js]: `schemaRefs.find` would be much simpler, it'll just give you `undefined` if nothing satisfies your function. Shouldn't even need to check `length`

**Pull request merged**